crates/turborepo-globwalk/src/lib.rs

```diff
@@ -832,8 +890,37 @@ mod test {
             "/repos/some-app/packages/left-pad/package.json",
             "/repos/some-app/packages/xzibit/package.json",
             "/repos/some-app/packages/xzibit/packages/yo-dawg/package.json",
-        ]
+        ],
+        Default::default()
         ; "nested packages work")]
+    #[test_case(&[
+            "/external/file.txt",
+            "/repos/some-app/apps/docs/package.json",
+            "/repos/some-app/apps/web/package.json",
+            "/repos/some-app/bower_components/readline/package.json",
+            "/repos/some-app/examples/package.json",
+            "/repos/some-app/node_modules/gulp/bower_components/readline/package.json",
+            "/repos/some-app/node_modules/react/package.json",
+            "/repos/some-app/package.json",
+            "/repos/some-app/packages/xzibit/package.json",
+            "/repos/some-app/packages/xzibit/node_modules/street-legal/package.json",
+            "/repos/some-app/packages/xzibit/node_modules/paint-colors/package.json",
+            "/repos/some-app/packages/xzibit/packages/yo-dawg/package.json",
+            "/repos/some-app/packages/xzibit/packages/yo-dawg/node_modules/meme/package.json",
+            "/repos/some-app/packages/xzibit/packages/yo-dawg/node_modules/yo-dawg/package.json",
+            "/repos/some-app/packages/colors/package.json",
+            "/repos/some-app/packages/faker/package.json",
+            "/repos/some-app/packages/left-pad/package.json",
+            "/repos/some-app/test/mocks/spanish-inquisition/package.json",
+            "/repos/some-app/tests/mocks/spanish-inquisition/package.json",
+        ],
+        "/repos/some-app/",
+        &["**/package.json"],
```

Comment:
Can you change the test case to better match how this setting gets used in `boundaries` e.g. a glob of `**/*.js` and making sure it doesn't pick up `js` files in `/repos/some-app/node_modules/`, but still picks up JS files in `/repos/some-app/apps/web/src/index.js`